src/controller_examples/v_replica_set_controller/proof/liveness/resource_match.rs

```diff
@@ -0,0 +1,482 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use crate::external_api::spec::*;
+use crate::kubernetes_api_objects::spec::{
+    api_method::*, common::*, dynamic::*, owner_reference::*, prelude::*, resource::*,
+};
+use crate::kubernetes_cluster::spec::{
+    builtin_controllers::types::BuiltinControllerChoice,
+    cluster::*,
+    cluster_state_machine::Step,
+    controller::types::{ControllerActionInput, ControllerStep},
+    message::*,
+};
+use crate::temporal_logic::{defs::*, rules::*};
+use crate::vstd_ext::{map_lib::*, string_view::*};
+use crate::v_replica_set_controller::{
+    model::{reconciler::*},
+    proof::{helper_invariants, predicate::*},
+    trusted::{spec_types::*, step::*, liveness_theorem::*},
+};
+use vstd::{prelude::*, string::*, math::abs};
+
+verus! {
+
+pub proof fn lemma_from_diff_and_init_to_current_state_matches(
```

Comment:
@codyjrivera This proof function is a bit too long. I suggest we tear it into a few smaller ones.